x-pack/plugins/enterprise_search/common/connectors/native_connectors.ts

```diff
@@ -1626,6 +1626,32 @@ export const NATIVE_CONNECTOR_DEFINITIONS: Record<string, NativeConnector | unde
         validations: [],
         value: false,
       },
+      use_document_level_security: {
+        default_value: false,
+        depends_on: [],
+        display: DisplayType.TOGGLE,
+        label: i18n.translate(
+          'xpack.enterpriseSearch.nativeConnectors.sharepoint_online.configuration.useDocumentLevelSecurityLabel',
+          {
+            defaultMessage: 'Use document level security',
+          }
+        ),
+        options: [],
+        order: 7,
+        required: true,
+        sensitive: false,
+        tooltip: i18n.translate(
+          'xpack.enterpriseSearch.nativeConnectors.sharepoint_online.configuration.useDocumentLevelSecurityTooltip',
+          {
+            defaultMessage:
+              'This will add access control data to content documents and will also sync identities with their corresponding access control to a separate index .search-acl-filter-index-name',
```

Comment:
I would leave out the name of the specific index, as we want to discourage direct index of indices like this.

Reply:
@leemthompo have a suggestion for the phrasing?

Reply:
@sphilipse @timgrein we had a very similar PR yesterday, I suggest we synchronize our approach there